src/Sentry/SentryStructuredLogger.cs

```diff
@@ -0,0 +1,212 @@
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Info, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Warning"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogWarning(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Warning, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Error"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogError(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Error, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Fatal"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogFatal(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Fatal, template, parameters, configureLog);
+        }
+    }
+
+    private void CaptureLog(SentryLogLevel level, string template, object[]? parameters, Action<SentryLog>? configureLog)
+    {
+        Debug.Assert(_options is not null);
+
+        var timestamp = _clock.GetUtcNow();
+
+        if (!TryGetTraceId(_hub, _scopeManager, out var traceId))
+        {
+            _options.DiagnosticLogger?.LogWarning("TraceId not found");
+        }
+
+        _ = TryGetParentSpanId(_hub, _scopeManager, out var parentSpanId);
+
+        var message = string.Format(CultureInfo.InvariantCulture, template, parameters ?? []);
+        SentryLog log = new(timestamp, traceId, level, message)
+        {
+            Template = template,
+            Parameters = ImmutableArray.Create(parameters),
+            ParentSpanId = parentSpanId,
+        };
+        log.SetAttributes(_options);
+
+        try
+        {
+            configureLog?.Invoke(log);
+        }
+        catch (Exception e)
+        {
+            //TODO: change to Diagnostic Logger (if enabled)
+            // see https://github.com/getsentry/sentry-dotnet/issues/4132
+            Console.WriteLine(e);
+            return;
```

Comment:
this is prob the safer approach, which ideally is documented under the configureLog xml doc and sentry docs

Reply:
Added XML doc comments to all `Log{Level}` messages and `SentryOptions.SetBeforeSendLog`, indicating that the log will not be captured when an `Exception` occurs during callback invocation.
Also changed to log an `Error` via the `IDiagnosticLogger` (if enabled).